apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.ts

```diff
@@ -205,6 +256,19 @@ router.post(
         res.locals.authn_user.user_id,
       );
       res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'regenerate_instance') {
+      if (!canDeleteAssessmentInstance(res.locals)) {
+        throw new error.HttpStatusError(403, 'Access denied');
+      }
+
+      await assessment.deleteAssessmentInstance(
+        res.locals.assessment.id,
+        res.locals.assessment_instance.id,
+        res.locals.authn_user.user_id,
+      );
+
+      flash('success', 'Regenerated your assessment instance.');
```

Comment:
There are some situations where this regeneration will not actually take place at this point. Three come to mind:
1. Group assessments (the assessment page will show the controls to manage the group);
2. Assessments with multiple instances (the assessment page will show a "Start assessment" button);
3. In "view student" (with access restrictions) if the assessment is not active (the assessment will not be created).

I think it's probably good to avoid a "Regenerated" wording here, at least in this cases.

Reply:
What wording do you propose instead?